### PR TITLE
fix(menu): rtl fix for menu

### PR DESCRIPTION
**Pull request by sijav**
#### Short description of what this resolves:
This will fix the menu for rtl direction.

#### Changes proposed in this pull request:

- fix min, max values for RTL (it should be exact opposite than LTR)
- fix the left right in scss (when we say right it must be stay as right)

**Ionic Version**: 3.x

**Fixes**: #11211 

**This will complete after**: #11822

### CODE REVIEW

**Comment by AmitMY** [comment]
Add "// scss-lint:disable PropertySpelling" above 'left: auto' so it will pass linting 
(Left and right are not allowed to be used except for special cases like this and like fab, this comment makes this a special case)

**Comment by AmitMY** [comment]
Remove the css changes, as they are conflicting, and this will be reviewed soon

**Comment by sijav** [comment]
@AmitMY what multidir mixin does? It just use the same code for both ltr and rtl? Can I use it in my other PRs?

**Comment by AmitMY** [comment]
`multi-dir` makes it: 
```
[dir="ltr"] selector, [dir="rtl"] selector {
  // something that is direction sensitive
}
```

Otherwise, specificity rules can override it.

You can and should use it in your PRs, only on directional properties. I have a list of all PRs and correct order to review them, and what relies on what. Your other PRs show as "not ready" for me as of a small dependency. I will update you as soon as that is taken care of

Left side is without it, middle is with, right is differences: http://ionic-snapshot-go.appspot.com/ionic2/snapshots/d4t/hhp/chrome_400x800

**Comment by sijav** [comment]
@AmitMY waiting for another review, I've tested `start` and `end` and it worked great as well.

File: src/components/menu/menu.scss [hunk 1]
```diff
@@ -38,6 +36,12 @@ ion-menu.show-menu {
   transform: translate3d(-9999px, 0, 0);
 
   contain: strict;
+
+  // scss-lint:disable PropertySpelling
+  @include multi-dir() {
+    left: 0;
+    right: auto;
+  }
```
Comment (AmitMY) [inline src/components/menu/menu.scss]: I did goof - the comment should be inside the mixin like - https://github.com/ionic-team/ionic/pull/11953

**Comment by AmitMY** [comment]
This goes well with https://github.com/ionic-team/ionic/pull/11336, which is not done becasue of bad animation after changing direction. Mind having a look? (`updatePosition` method is what changes the animation, but it breaks it if it runs more than once)

File: src/components/menu/menu.scss (hunk 1 above)
Comment (AmitMY) [inline src/components/menu/menu.scss]: Leave `menu.scss` unchanged
The correct default is `start`, and thus `0, auto, 0, 0` is correct
You are correct that there is a need to address left, so https://github.com/ionic-team/ionic/pull/11953

Sorry for all of the mess around this.

Comment (sijav) [inline src/components/menu/menu.scss]: Oh ... alright

Comment (sijav) [inline src/components/menu/menu.scss]: @AmitMY but without this left menu open at a wrong position especially in tablets... mind check it?

**Comment by sijav** [comment]
@AmitMY I reverted the commit but if you mind please check the left menu in tablet mode it doesn't open at an appropriate position also about the commit I'll take a look but there's a lot of things changed which I'm not sure why are these things changed? why we are using a custom animation here?

**Comment by AmitMY** [comment]
@sijav I added an explenation to the body of that PR, as it is more relevant there. About left menu, I think it is fixed with https://github.com/ionic-team/ionic/pull/11953

File: src/components/menu/menu-gestures.ts [hunk 2]
```diff
@@ -57,14 +57,14 @@ export class MenuContentGesture extends SlideEdgeGesture {
   }
 
   onSlide(slide: SlideData, ev: any) {
-    const z = (this.menu.isRightSide ? slide.min : slide.max);
+    const z = ((this.menu.isRightSide && !this.plt.isRTL) || (!this.menu.isRightSide && this.plt.isRTL) ? slide.min : slide.max);
```
Comment (AmitMY) [inline src/components/menu/menu-gestures.ts]: `(this.menu.isRightSide && !this.plt.isRTL) || (!this.menu.isRightSide && this.plt.isRTL)`
is logically equal to:
`this.menu.isRightSide !== this.plt.isRTL`

Can you fix this in all of the expressions in this PR?

Comment (sijav) [inline src/components/menu/menu-gestures.ts]: well technically yes but I put it that way for more readability ...
Ok I will change all of em

**Comment by sijav** [comment]
I can confirm that menu scss problems has been fixed by #11953 Thanks @AmitMY

File: src/components/menu/menu-gestures.ts [hunk 3]
```diff
@@ -89,7 +89,7 @@ export class MenuContentGesture extends SlideEdgeGesture {
 
   getElementStartPos(slide: SlideData, ev: any) {
     const menu = this.menu;
-    if (menu.isRightSide) {
+    if (this.menu.isRightSide !== this.plt.isRTL) {
```
Comment (AmitMY) [inline src/components/menu/menu-gestures.ts]: Can you please change `this.menu` to `menu`?

Comment (sijav) [inline src/components/menu/menu-gestures.ts]: @AmitMY yeah sure sorry my bad

Comment (AmitMY) [inline src/components/menu/menu-gestures.ts]: I missed it too :) orders from above

**Pull request merged**